tests/test_psfCandidate.py

```diff
@@ -37,46 +36,96 @@
     display = False
 
 
+def makeEmptyCatalog(psfCandidateField=''):
+    """Return an empty catalog with a useful schema for psfCandidate testing.
+
+    Parameters
+    ----------
+    psfCandidateField : `str`
+        The name of a flag field to add to the schema.
+
+    Returns
+    -------
+    catalog : `lsst.afw.table.SourceCatalog`
+        The created (empty) catalog.
+    """
+    schema = afwTable.SourceTable.makeMinimalSchema()
+    lsst.afw.table.Point2DKey.addFields(schema, "centroid", "centroid", "pixels")
+    if psfCandidateField != '':
+        schema.addField(psfCandidateField, type="Flag", doc="Is a psfCandidate?")
+    catalog = afwTable.SourceCatalog(schema)
+    catalog.defineCentroid('centroid')
+
+    return catalog
+
+
+def createFakeSource(x, y, catalog, exposure, threshold=0.1):
+    """Create a fake source at the given x/y centroid location.
+
+    Parameters
+    ----------
+    x,y : `int`
+        The x and y centroid coordinates to place the image at.
+    catalog : `lsst.afw.table.SourceCatalog`
+        The catalog to add the new source to.
+    exposure : `lsst.afw.image.Exposure`
+        The exposure to add the source to.
+    threshold : `float`, optional
+        The footprint threshold for identifying the source.
+
+    Returns
+    -------
+    source : `lsst.afw.table.SourceRecord`
+        The created source record that was added to ``catalog``.
+    """
+    source = catalog.addNew()
+    source['centroid_x'] = x
+    source['centroid_y'] = y
+
+    exposure.getMaskedImage().getImage()[x, y] = 1.0
+    fpSet = afwDet.FootprintSet(exposure.getMaskedImage(), afwDet.Threshold(threshold), "DETECTED")
+    if display:
+        ds9.mtv(exposure, frame=1)
+        for fp in fpSet.getFootprints():
+            for peak in fp.getPeaks():
+                ds9.dot("x", peak.getIx(), peak.getIy(), frame=1)
+
+    # There might be multiple footprints; only the one around x,y should go in the source
+    found = False
+    for fp in fpSet.getFootprints():
+        if fp.contains(afwGeom.Point2I(x, y)):
+            found = True
+            print(fp.getBBox())
+            break
+    assert found, "Unable to find central peak in footprint: faulty test"
```

Comment:
It's not immediately clear that this line is asserting `found` must be `True` and prints the "unable to find..." message if it's False. But if this is the correct/pythonic way to do so, I can live with it.

Reply:
This is the syntax for the python `assert` statement. Is the purpose of the assert here clear?

Would you be happier with `assert found is True, "sometext"` instead?

Reply:
I think it's really a question of it looking different because everywhere else in tests you see `self.assertTrue(found)` but here you see a bare assert because `self` has not been passed in to the function.

Reply:
If this was in a class that the other test classes inherited from (maybe with multiple inheritance) then the standard scheme could be used.

Reply:
In this case, it's supposed to be "is my test even sane?", hence the raw assert.

Reply:
I've added a note to clarify the purpose.

Reply:
Please don't use `foo is True`; it isn't the same as testing the boolean value of `foo` and it's not good Python unless you are sure you need exactly the `True` singleton.

Reply:
Oh, right @r-owen I forgot about that. Anyway, I think it should be clear with the comment I added.